### PR TITLE
stats: setting default value for returned values

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: stats: setting default value for returned values
Additional Description:
Our internal msan test fails here, because the 2 data-members could be directly fetched without these values being explicitly initialized. Initializing the values solves the issue.
There should not be any impact, as the explicit values are the also the default ones.

Risk Level: Low
Testing: N/A.
Docs Changes: N/A.
Release Notes: N/A.
Platform Specific Features: N/A.

Signed-off-by: Adi Suissa-Peleg <adip@google.com>

### CODE REVIEW

**Review by KBaichoo — DISMISSED** [review]

File: source/common/stats/histogram_impl.h [hunk 1]
```diff
@@ -68,8 +68,8 @@ class HistogramStatisticsImpl final : public HistogramStatistics, NonCopyable {
   ConstSupportedBuckets& supported_buckets_;
   std::vector<double> computed_quantiles_;
   std::vector<uint64_t> computed_buckets_;
-  uint64_t sample_count_;
-  double sample_sum_;
+  uint64_t sample_count_{0};
+  double sample_sum_{0};
```
Comment (jmarantz) [inline source/common/stats/histogram_impl.h:72]: should we also init unit_  if we are initing the other scalars?

Comment (pradeepcrao) [reply source/common/stats/histogram_impl.h:72]: It is initialized in member initializer list of both constructors, but doing it here makes it easier to see the default value.

Comment (adisuissa) [reply source/common/stats/histogram_impl.h:72]: I don't think it can be initialized here as it is defined as `const`.

Comment (adisuissa) [reply source/common/stats/histogram_impl.h:72]: I can move the initialization of `sample_count_` and `sample_sum_` to the MIL, if you prefer.

Comment (KBaichoo) [reply source/common/stats/histogram_impl.h:72]: I think you can add a default here even though it's const.

Comment (adisuissa) [reply source/common/stats/histogram_impl.h:72]: TIL, thanks!
Done.

**Review by KBaichoo — APPROVED** [review]

**Review by alyssawilk — APPROVED** [review]

**Review by jmarantz — APPROVED** [review]

**Pull request merged**